GMPWrapper/scripts/convertMarlinSteeringToGaudi.py

```diff
@@ -55,6 +56,64 @@ def getProcessors(tree):
   return processors
 
 
+# Replace with constants if pattern found in value(s)
+def replaceConstants(value, constants):
+  formatted_array = []
+  split_values = value.split()
+  for val in split_values:
+    captured_patterns = re.findall('\$\{\w*\}', val)
+    if len(captured_patterns) == 0:
+      formatted_array.append("\"{}\"".format(val))
+    elif len(captured_patterns) >= 1:
+      val_format = re.sub(r'\$\{(\w*)\}', r'%(\1)s', val)
+      val_format = "\"{}\" % CONSTANTS".format(val_format)
+      formatted_array.append(val_format)
+
+  return ", ".join(formatted_array)
+
+
+# Find constant tags, write them to python and replace constants within themselves
+def convertConstants(lines, tree):
+  constants = dict()
+
+  constElements = tree.findall('constants/include')
+  if constElements:
+    print('WARNING: include statements found inside constants tag')
+
+  constElements = tree.findall('constants/constant')
+  for const in constElements:
+    constants[const.attrib.get('name')] = getValue(const)
+
+  for key,value in constants.items():
+    if value:
+      formatted_array = []
+      split_values = value.split()
+      if len(split_values) == 1:
+        captured_patterns = re.findall('\$\{\w*\}', value)
+        for pattern in captured_patterns:
+          constants[key] = re.sub(r'\$\{(\w*)\}', r'%(\1)s', constants[key])
+        constants[key] = "\"{}\"".format(constants[key])
+      elif len(split_values) > 1:
+        for val in split_values:
+          captured_patterns = re.findall('\$\{\w*\}', val)
+          if len(captured_patterns) == 0:
+            formatted_array.append("\"{}\"".format(val))
+          elif len(captured_patterns) >= 1:
+            val_format = re.sub(r'\$\{(\w*)\}', r'%(\1)s', val)
+            val_format = "\"{}\"".format(val_format)
+            formatted_array.append(val_format)
+        constants[key] = '[{}]'.format(", ".join(formatted_array))
+
+  lines.append("\nCONSTANTS = {")
+  for key in constants:
+    lines.append("\t\t'{}': {},".format(key, constants[key]))
```

Comment:
Is the rest of the code also using tabs for whitespace in the output file?
```python
' ' * len('CONSTANTS = {') + "{}: {},".format...
```
or something

Reply:
No. The plan was to introduce this in the rest of the code, but for now I'll remove it as it makes it inconsistent.